src/charts/c3/c3-chart.directive.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.charts.directive:pfC3Chart
+ *
+ * @description
+ *   Directive for wrapping c3 library
+ *
+ *   Note: The 'patternfly.charts' module is not a dependency in the default angular 'patternfly' module.
+ *   In order to use patternfly charts you must add 'patternfly.charts' as a dependency in your application.
+ *
+ *
+ * @param {string} id the ID of the container that the chart should bind to
+ * @param {expression} config the c3 configuration options for the chart
+ *
+ * @example
+ <example module="patternfly.charts">
+<file name="index.html">
+ <div ng-controller="ChartCtrl">
+   <div pf-c3-chart id="chartId"  config="chartConfig"></div>
+
+   <form role="form" style="width:300px">
+     Total = {{total}}, Used = {{used}}, Available = {{available}}
+     <div class="form-group">
+       <label>Used</label>
+       <input type="text" class="form-control" ng-model="newUsed">
+     </div>
+     <input type="button" ng-click="submitform(newUsed)" value="Go" />
+   </form>
+ </div>
+ </file>
+
+<file name="script.js">
+ function ChartCtrl($scope) {
+    $scope.used = 950;
+    $scope.total = 1000;
+    $scope.available =  $scope.total - $scope.used;
+
+    $scope.chartConfig = {"donut":
+                     {"title":"MHz Used",
+                      "label":{"show":false},
+                      "width":10
+                     },
+                     "size":{"height":130},
+                     "legend":{"show":false},
+                     "color":{"pattern":["#0088CE","#D1D1D1"]},
+                     "tooltip":{},
+                     "data":{"columns":[["Used","950"],["Available",50]],
+                     "type":"donut",
+                     "donut":{
+                         "label":{"show":false}
+                      },
+                      "groups":[["used","available"]],
+                      "order":null
+                      }
+                  };
+
+    $scope.updateAvailable = function(val){
+      $scope.available =  $scope.total - $scope.used;
+    }
+
+    $scope.submitform = function(val){
+      $scope.used = val;
+      $scope.updateAvailable();
+      $scope.chartConfig.data.columns = [["Used",$scope.used],["Available",$scope.available]];
+   };
+ }
+ </file>
+ </example>
+ */
+(function(c3){
+  'use strict';
+
+  angular.module('patternfly.charts')
+  .directive('pfC3Chart', ['$timeout', function($timeout) {
+
+    return {
+      restrict: 'A',
+      scope: {
+        config: '='
+      },
+      template: '<div id=""></div>',
+      replace: true,
+      link: function(scope, element, attrs) {
+        scope.$watch('config', function() {
+          $timeout(function () {
```

Comment:
I removed the $timeout, and put in 'scope.$apply()', ex: 

link: function(scope, element, attrs) {
        scope.$watch('config', function() {
          //generate c3 chart data
          var chartData = scope.config;
          chartData.bindto = '#' + attrs.id;
          c3.generate(chartData);
          scope.$apply();
        },true);
      }

I get:
"Error: url or json or rows or columns is required.
	    at /home/dtaylor/Dev/repos/angular-patternfly/lib/patternfly/components/c3/c3.js:92"

If I put scope.$apply() before the c3.generate, ex:

link: function(scope, element, attrs) {
        scope.$watch('config', function() {
          //generate c3 chart data
          var chartData = scope.config;
          chartData.bindto = '#' + attrs.id;
          scope.$apply();
          c3.generate(chartData);
        },true);
      }

I get: 
"Error: [$rootScope:inprog] $digest already in progress
	http://errors.angularjs.org/1.3.17/$rootScope/inprog?p0=%24digest"

Not sure how to do it without the $timeout, think it is needed since we are wrapping the c3 lib.